it/controllers/pensions/incomeFromPensions/UkPensionIncomeSummaryControllerISpec.scala

```diff
@@ -0,0 +1,179 @@
+    val expectedCaption: Int => String = (taxYear: Int) => s"Income from pensions for 6 April ${taxYear - 1} to 5 April $taxYear"
+    val expectedButtonText = "Continue"
+    val expectedTitle = "UK pension income"
+    val expectedHeading = "UK pension income"
+    val change = "Change"
+    val remove = "Remove"
+    val expectedAddAnotherText = "Add another pension scheme"
+    val expectedAddPensionSchemeText = "Add a pension scheme"
+  }
+
+  object CommonExpectedCY extends CommonExpectedResults {
+    val expectedCaption: Int => String = (taxYear: Int) => s"Income from pensions for 6 April ${taxYear - 1} to 5 April $taxYear"
+    val expectedButtonText = "Continue"
+    val expectedTitle = "UK pension income"
+    val expectedHeading = "UK pension income"
+    val change = "Change"
+    val remove = "Remove"
+    val expectedAddAnotherText = "Add another pension scheme"
+    val expectedAddPensionSchemeText = "Add a pension scheme"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, Nothing]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY)
+  )
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        import Selectors._
+        import user.commonExpectedResults._
+
+        val pensionName1 = anUkPensionIncomeViewModelOne.pensionSchemeName.get
+        val pensionName2 = anUkPensionIncomeViewModelTwo.pensionSchemeName.get
+
+        "render the 'UK pension income' summary list page with pre-filled content" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropPensionsDB()
+            val viewModel = anIncomeFromPensionsViewModel.copy(uKPensionIncomes = Seq(anUkPensionIncomeViewModelOne, anUkPensionIncomeViewModelTwo))
+            insertCyaData(pensionsUserDataWithIncomeFromPensions(viewModel), aUserRequest)
+            urlGet(fullUrl(ukPensionSchemeSummaryListUrl(taxYearEOY)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(expectedTitle)
+          h1Check(expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(pensionName1, pensionNameSelector(1))
+          textOnPageCheck(pensionName2, pensionNameSelector(2))
+          
+          //TODO: replace hrefs "#" below with link to first details page when available .e.g. PensionSchemeDetailsController.show(taxYearEOY, Some(1)).url
```

Comment:
I think the change link has been changed to redirect to the pension specific CYA (slide 10 on the design slides), could you update the todos?

Reply:
ok